decryptor/postgresql/utils.go

```diff
@@ -176,6 +189,142 @@ func (p *BindPacket) Zeroize() {
 	}
 }
 
+// GetParameters extracts statement parameters from Bind packet.
+func (p *BindPacket) GetParameters() ([]base.BoundValue, error) {
+	values := make([]base.BoundValue, len(p.paramValues))
+	for i := range values {
+		encoding, err := p.parameterEncoding(i)
+		if err != nil {
+			return nil, err
+		}
+		values[i] = base.NewBoundValue(p.paramValues[i], encoding)
+	}
+	return values, nil
+}
+
+func (p *BindPacket) parameterEncoding(i int) (base.BoundValueEncoding, error) {
+	// See "Bind" description in https://www.postgresql.org/docs/current/protocol-message-formats.html
+	// If there are no formats then all values use the default: text.
+	if len(p.paramFormats) == 0 {
+		return base.BindText, nil
+	}
+	// If there is only one format then it is used for all values.
+	var format uint16
+	if len(p.paramFormats) == 1 {
+		format = p.paramFormats[0]
+	} else {
+		format = p.paramFormats[i]
+	}
+	// Options currently include text and binary formats.
+	switch format {
+	case bindFormatText:
+		return base.BindText, nil
+	case bindFormatBinary:
+		return base.BindBinary, nil
+	default:
+		log.WithField("index", i).WithField("format", format).Debug("Unknown Bind format")
+		return base.BindText, ErrUnknownFormat
+	}
+}
+
+// SetParameters updates statement parameters from Bind packet.
+func (p *BindPacket) SetParameters(values []base.BoundValue) {
+	p.updateParameterFormats(values)
```

Comment:
can we add one more function like `updateParameters(values)` which will do both functions? as I understand, formats and values must be synchronized with each other and it should be done together, yes?

Reply:
> can we add one more function like updateParameters(values) which will do both functions?

Uh... `SetParameters` is this function.

> formats and values must be synchronized with each other and it should be done together

Yeah, that's true.

I've split it into two functions early on in attempt to improve readability. However, this does not seem to improve it that much, so I guess it's better to just put everything into `SetParameters`.